### PR TITLE
More error checking in system.c (fixes #1317)

### LINKED ISSUE #1317: Output better error messages
> Outputting useful error messages is also an area in which Teeworlds/DDNet could improve.

Per @heinrich5991@#1302

Especially in `system.c` we throw out most if not all errors. Would be useful to output errors from standard functions. This would also litter the console output a bit so would have to be implemented along with a verbosity level.

### PR DESCRIPTION
**Pull request by def-**
I probably didn't catch all interesting locations, but it's a start.

### CODE REVIEW

**Comment by heinrich5991** [comment]
`strerror` is thread-unsafe according to its man page.

**Comment by Learath2** [comment]
Could have `strerror` behind a mutex or use `strerror_r` but thats POSIX.

**Comment by def-** [comment]
I really don't want potential lags because we're locking for some error logging that might turn out to be not a problem.

File: src/game/server/score/sql_score.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* (c) Shereef Marzouk. See "licence DDRace.txt" and the readme.txt in the root of the distribution for more information. */
 /* Based on Race mod stuff and tweaked by GreYFoX@GTi and others to fit our DDRace needs. */
-/* CSqlScore class by Sushi */
+/* CSqlScoreclass by Sushi */
```
Comment (heinrich5991) [inline src/game/server/score/sql_score.cpp]:
```suggestion
/* CSqlScore class by Sushi */
```
typo

File: src/base/system.c [hunk 2]
```diff
@@ -901,7 +949,11 @@ int64 time_get_impl()
 		return last;
 #elif defined(CONF_FAMILY_UNIX)
 		struct timespec spec;
-		clock_gettime(CLOCK_MONOTONIC, &spec);
+		if(clock_gettime(CLOCK_MONOTONIC, &spec) != 0)
+		{
+			dbg_msg("clock", "gettime failed: %d", errno);
+			return 0;
+		}
```
Comment (heinrich5991) [inline src/base/system.c]: Modifying behavior here, but would probably not have worked with the old code either, so it's fine.

File: src/base/system.c [hunk 3]
```diff
@@ -1296,7 +1349,7 @@ static int priv_net_create_socket(int domain, int type, struct sockaddr *addr, i
 			buf[0] = 0;
 		dbg_msg("net", "failed to create socket with domain %d and type %d (%d '%s')", domain, type, error, buf);
 #else
-		dbg_msg("net", "failed to create socket with domain %d and type %d (%d '%s')", domain, type, errno, strerror(errno));
+		dbg_msg("net", "failed to create socket with domain %d and type %d (%d)", domain, type, errno);
```
Comment (heinrich5991) [inline src/base/system.c]: Slight degradation in usability (think "addr already in use").

**Comment by Learath2** [comment]
You seem to have made some git mistake, this includes 0d5db8a, but we already merged an equivalent commit that you had modified by force pushing before.

**Comment by def-** [comment]
Oh, weird, must have been my squash.

**Pull request merged**